src/shared/components/case-editor/services/cases.service.ts

```diff
@@ -217,15 +218,94 @@ export class CasesService {
     if (!eventTrigger.wizard_pages) {
       eventTrigger.wizard_pages = [];
     }
-    /* FIXME: find a better place for this code */
+
     eventTrigger.wizard_pages.forEach((wizardPage: WizardPage) => {
       wizardPage.parsedShowCondition = new ShowCondition(wizardPage.show_condition);
-      let orderedWPFields = this.orderService.sort(wizardPage.wizard_page_fields);
-      wizardPage.case_fields = orderedWPFields.map((wizardField: WizardPageField) => {
-        let case_field = eventTrigger.case_fields.find(cf => cf.id === wizardField.case_field_id);
-        case_field.wizardProps = wizardField;
-        return case_field;
+      wizardPage.case_fields = wizardPage.wizard_page_fields.map((wizardField: WizardPageField) => {
+        return this.wizardPageFieldToCaseFieldMapper(wizardField, eventTrigger.case_fields);
+      });
+      wizardPage.case_fields = this.deepOrder(wizardPage.case_fields);
+    });
+  }
+
+  private wizardPageFieldToCaseFieldMapper(wizardPageField: WizardPageField, caseFields: CaseField[]): CaseField {
+
```

Comment:
@rafalkalita we have introduced some complexity to this Service which was already enough complex. We should move this new logic 
into its own class.
Also we need to simplify these methods as well, too long sequences of if else... perhaps doing some extract methods with some self 
explanatory method names would help